main.go

```diff
@@ -108,15 +119,22 @@ func main() {
 		http.Handle(cfg.telemetryPath, promhttp.Handler())
 
 		logger := cfg.createLogger()
-		awsQueryConfigs := cfg.buildAWSConfig()
-		awsWriteConfigs := cfg.buildAWSConfig()
 
-		timestreamClient := timestream.NewBaseClient(cfg.defaultDatabase, cfg.defaultTable)
+		ctx := context.Background()
+		awsQueryConfigs, err := cfg.buildAWSConfig(ctx, cfg.maxRetries)
+		if err != nil {
+			timestream.LogError(logger, "Failed to build AWS configuration for query", err)
+			os.Exit(1)
+		}
 
-		awsQueryConfigs.MaxRetries = aws.Int(cfg.maxRetries)
-		timestreamClient.NewQueryClient(logger, awsQueryConfigs)
+		awsWriteConfigs, err := cfg.buildAWSConfig(ctx, writeClientMaxRetries)
```

Comment:
Is there any reason the writeClientMaxRetries is hard-coded but the maxRetries isn't for the query client? If not I think we should add a flag and make query and write retries configurable.